File: JS/workbook2/party.js
"use strict";

const fs = require("fs");

const input =
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString().trim().split("\n")
    : `5 10
0 4 4 8 7
7 0 7 7 4
1 4 0 5 4
5 2 2 0 7
1 4 1 6 0
1 3 8
2 4 1
4 1 1
1 5 5
3 2 1
3 2 5
4 5 10
5 3 2
1 4 1
1 4 11`.split("\n");

const [n, m] = input[0].split(" ").map((v) => parseInt(v));
const board = input
  .slice(1, n + 1)
  .map((v) => v.split(" ").map((value) => parseInt(value)));

for (let i = 0; i < n; i++) {
  for (let j = 0; j < n; j++) {
    for (let k = 0; k < n; k++) {
      if (j === k) continue;

      board[j][k] = Math.min(board[j][k], board[j][i] + board[i][k]);
    }
  }
}

input.slice(n + 1).forEach((v) => {
  const [start, end, time] = v.split(" ").map((v) => parseInt(v));

  if (board[start - 1][end - 1] <= time) console.log("Enjoy other party");
  else console.log("Stay here");
});

File: src/workbook2/Party.java
package workbook2;

import java.io.*;

public class Party {
    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] nm = br.readLine().split(" ");
        int n = Integer.parseInt(nm[0]), m = Integer.parseInt(nm[1]);

        int[][] board = new int[n][n];

        for (int i = 0; i < n; i++) {
            String[] s = br.readLine().split(" ");
            for (int j = 0; j < n; j++) {
                board[i][j] = Integer.parseInt(s[j]);
            }
        }

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                for (int k = 0; k < n; k++) {
                    if(j == k) continue;

                    board[j][k] = Math.min(board[j][k], board[j][i] + board[i][k]);
                }
            }
        }

        for (int i = 0; i < m; i++) {
            String[] s = br.readLine().split(" ");
            int start = Integer.parseInt(s[0]), end = Integer.parseInt(s[1]), time = Integer.parseInt(s[2]);

            if(board[start - 1][end - 1] <= time) bw.write("Enjoy other party\n");
            else bw.write("Stay here\n");
        }

        bw.flush();
        bw.close();
    }
}
